src/graph/dg.rs

```diff
@@ -374,71 +374,191 @@ pub fn resolve_all_paths_only_of_references(
+//     stack_graph: &StackGraph
+// ) {
+//     let mut references = Vec::<Handle<Node>>::new();
+//     log::info!("finding references");
+//     for node_handle in stack_graph.iter_nodes() {
+//         match stack_graph.index(node_handle) {
+//             PushSymbol(_) => {
+//                 if stack_graph.index(node_handle).is_reference() {
+//                     let refkind = Refkind::from(find_debug_info(
+//                         stack_graph,
+//                         node_handle,
+//                         "refkind".to_string()
+//                     ).unwrap_or_default());
+//                     match refkind.is_nothing() {
+//                         false => {
+//                             references.push(node_handle);
+//                         },
+//                         true => (),
+//                     }
+//                 }
+//             },
+//             _ => {}
+//         }
+//     }
+//     log::info!("found {} references", references.len());
+    
+//     let mut bindings = 0;
+//     let progress_bar = indicatif::ProgressBar::new(references.len().try_into().unwrap());
+//     for node_handle in references {
+//         let mut paths = Paths::new();
+//         let mut cycle_detector = crate::graph::lavatrice::Lavatrice::new();
+//         let mut queue = [node_handle].iter()
+//             .into_iter()
+//             .filter_map(|node| Path::from_node(stack_graph, &mut paths, *node))
+//             .collect::<VecDeque<_>>();
+//         while let Some(path) = queue.pop_front() {
+//             NoCancellation.check("finding paths").unwrap();
+//             if !cycle_detector.should_process_path(&path, |probe| probe.cmp(stack_graph, &mut paths, &path)) {
+//                 continue;
+//             }
+//             if path.is_complete(stack_graph) {
+//                 match Defkind::from(
+//                     find_debug_info(
+//                         stack_graph,
+//                         path.end_node,
+//                         "defkind".to_string()
+//                     ).unwrap_or_default()
+//                 ).is_nothing() {
+//                     true => {},
+//                     false => {
+//                         if explorer.name_bindings.get(&path.start_node).is_none() {
+//                             bindings += 1;
+//                             progress_bar.inc(1);
+//                             explorer.set_name_binding(path.start_node, path.end_node);
+//                             break;
+//                         } else {
+//                             log::warn!("found a duplicate for a name binding")
+//                         }
+//                     }
+//                 }
+//             }
+//             path.extend(stack_graph, &mut paths, &mut queue);
+//         }
+//     }
+//     progress_bar.finish();
+//     log::info!("found {} bindings", bindings);
+// }
+
+pub fn resolve_all_paths_reference_by_reference(
```

Comment:
This one resolves references one by one, instead of all together, which helps a lot with memory usage.